src/commands/Minion/alch.ts

```diff
@@ -0,0 +1,159 @@
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			cooldown: 1,
+			usage: '<quantity:int{1}|itemName:...string> [itemName:...string]',
+			usageDelim: ' '
+		});
+	}
+
+	async run(msg: KlasaMessage, [quantity, itemName]: [number | null | string, string]) {
+		if (typeof quantity === 'string') {
+			itemName = quantity;
+			quantity = null;
+		}
+
+		const { id: itemId, highalch, name } = getOSItem(itemName);
+		const numOfItems = await msg.author.numberOfItemInBank(itemId);
+		// 5 tick action
+		const timePerAlch = Time.Second * 3;
+
+		if (numOfItems === 0) throw `'You don't have any of this item to alch'`;
+
+		if (!quantity) {
+			quantity = numOfItems;
+		}
+
+		const maxCasts = Math.min(Math.floor(msg.author.maxTripLength / timePerAlch), numOfItems);
+
+		if (quantity * timePerAlch > msg.author.maxTripLength) {
+			throw `The max number of alchs you can do is ${maxCasts}!`;
+		}
+
+		const duration = quantity * timePerAlch;
+		let fireRuneCost = quantity * 5;
+
+		const unlimitedFireRuneProviders = resolveItems([
+			'Staff of fire',
+			'Fire battlestaff',
+			'Mystic fire staff',
+			'Lava battlestaff',
+			'Mystic lava staff',
+			'Steam battlestaff',
+			'Mystic steam staff',
+			'Smoke battlestaff',
+			'Mystic smoke staff',
+			'Tome of fire'
+		]);
+
+		for (const runeProvider of unlimitedFireRuneProviders) {
+			if (
+				hasItemEquipped(runeProvider, msg.author.settings.get(UserSettings.Gear.Skilling))
+			) {
+				fireRuneCost = 0;
+				break;
+			}
+		}
+
+		const alchValue = quantity * highalch;
+
+		const consumedItems = resolveNameBank({
+			...(fireRuneCost > 0 ? { 'Fire rune': fireRuneCost } : {}),
+			'Nature rune': quantity,
+			[name]: quantity
+		});
+
+		const consumedItemsString = await createReadableItemListFromBank(
+			this.client,
+			consumedItems
+		);
+
+		const userBank = await msg.author.settings.get(UserSettings.Bank);
+
+		if (!bankHasAllItemsFromBank(userBank, consumedItems)) {
+			throw `You don't have the required items, you need ${consumedItemsString}`;
+		}
+
+		if (!msg.flagArgs.confirm && !msg.flagArgs.cf) {
+			const alchMessage = await msg.channel.send(
+				`${
+					msg.author
+				}, say \`confirm\` to alch ${quantity} ${name} for ${alchValue.toLocaleString()} (${Util.toKMB(
+					alchValue
+				)})`
+			);
+
+			try {
+				await msg.channel.awaitMessages(
+					_msg =>
+						_msg.author.id === msg.author.id &&
+						_msg.content.toLowerCase() === 'confirm',
+					options
+				);
+			} catch (err) {
+				return alchMessage.edit(`Cancelling alch of ${quantity}x ${name}`);
+			}
+		}
+
+		await msg.author.settings.update(
+			UserSettings.Bank,
+			removeBankFromBank(userBank, consumedItems)
+		);
+
+		const data: AlchingActivityTaskOptions = {
+			itemName: name,
+			userID: msg.author.id,
+			channelID: msg.channel.id,
+			quantity,
+			duration,
+			alchValue,
+			bryophytasStaffEquipped: hasItemEquipped(
```

Comment:
i dont think you need to snapshot the gear when sending out because minion gear should be locked while in an activity afaik